src/cmd/ksh93/bltins/typeset.c

```diff
@@ -1276,7 +1274,18 @@ static int unall(int argc, char **argv, register Dt_t *troot, Shell_t* shp)
 				}
 					
 				if(shp->subshell)
-					np=sh_assignok(np,0);
+				{
+					if(!nv_isattr(np,NV_NODISC|NV_ARRAY) && !nv_isvtree(np))
```

Comment:
Why are we doing this test when we're already know we're changing `PATHNOD` which we know has a discipline function? Genuine question, it is possible I am missing something here…